tun2socks/tun2socks.c

```diff
@@ -1429,9 +1429,26 @@ int process_device_udp_packet (uint8_t *data, int data_len)
     switch (ip_version) {
         case 4: {
             // ignore non-UDP packets
-            if (data_len < sizeof(struct ipv4_header) || data[offsetof(struct ipv4_header, protocol)] != IPV4_PROTOCOL_UDP) {
+            if (data_len < sizeof(struct ipv4_header)) {
                 goto fail;
             }
+            switch (data[offsetof(struct ipv4_header, protocol)]) {
+                case IPV4_PROTOCOL_UDP: break;
+                case 1: { // ICMPv4
```

Comment:
Please don't hard code numbers. Try to add macros here: https://github.com/shadowsocks/badvpn/blob/shadowsocks-android/misc/ipv4_proto.h

Reply:
#